src/routes/drive-planning.freights.tsx

```diff
@@ -13,10 +23,148 @@ export const Route = createFileRoute("/drive-planning/freights")({
 
 function DrivePlanningFreights() {
   const { t } = useTranslation();
+  const { freightsApi, sitesApi, freightUnitsApi } = useApi();
+
+  const [paginationModel, setPaginationModel] = useState<GridPaginationModel>({ page: 0, pageSize: 25 });
+  const [totalResults, setTotalResults] = useState(0);
+
+  const freightsQuery = useQuery({
+    queryKey: ["freights", paginationModel],
+    queryFn: async () => {
+      const [freights, headers] = await freightsApi.listFreightsWithHeaders({
+        ...ServerSidePaginationUtils.paginationModelToRange(paginationModel),
+      });
+      const count = parseInt(headers.get("x-total-count") ?? "0");
+      setTotalResults(count);
+      return freights;
+    },
+  });
+
+  const customerSitesQuery = useQuery({
+    queryKey: ["customerSites"],
+    queryFn: async () => await sitesApi.listSites(),
```

Comment:
No need to use async as the return type is promise
```suggestion
    queryFn: () => sitesApi.listSites(),
```

Reply:
True! Resolved.